python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -676,6 +691,42 @@ def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationRe
         return []
 
 
+@whitelist_for_serdes
+class SkipOnBackfillInProgressRule(
+    AutoMaterializeRule,
+    NamedTuple(
+        "_SkipOnBackfillInProgressRule", [("skip_all_partitions_of_backfilling_asset", bool)]
+    ),
+):
+    @property
+    def decision_type(self) -> AutoMaterializeDecisionType:
+        return AutoMaterializeDecisionType.SKIP
+
+    @property
+    def description(self) -> str:
+        return "asset is targeted by an in-progress backfill"
+
+    def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationResults:
+        backfill_in_progress_candidates: AbstractSet[AssetKeyPartitionKey] = set()
+
+        backfilling_subset = (
+            context.instance_queryer.get_active_backfill_target_asset_graph_subset()
+        )
+
+        for candidate in context.candidates:
```

Comment:
absolutely a nitpick, but I think we don't even need a loop here in the case that the boolean is set to True (we just add all candidates to the skip set of the asset key is in the backfilling subset), so I think we can shave off most of the lines of code here.